src/main/scala/vsys/transaction/contract/ExecuteContractFunctionTransaction.scala

```diff
@@ -0,0 +1,121 @@
+                                                      proofs: Proofs)
+  extends ProvenTransaction {
+
+  override val transactionType: TransactionType.Value = TransactionType.ExecuteContractFunctionTransaction
+
+  lazy val toSign: Array[Byte] = Bytes.concat(
+    Array(transactionType.id.toByte),
+    contractId.bytes.arr,
+    Shorts.toByteArray(funcIdx),
+    Deser.serializeArray(DataEntry.serializeArrays(data)),
+    BytesSerializable.arrayWithSize(attachment),
+    Longs.toByteArray(fee),
+    Shorts.toByteArray(feeScale),
+    Longs.toByteArray(timestamp)
+  )
+
+  override lazy val json: JsObject = jsonBase() ++ Json.obj(
+    "contractId" -> contractId.address,
+    "functionIndex" -> funcIdx,
+    "functionData" -> Base58.encode(DataEntry.serializeArrays(data)),
+    "attachment" -> Base58.encode(attachment),
+    "fee" -> fee,
+    "feeScale" -> feeScale,
+    "timestamp" -> timestamp
+  )
+
+  override val assetFee: (Option[AssetId], Long, Short) = (None, fee, feeScale)
+  override lazy val bytes: Array[Byte] = Bytes.concat(toSign, proofs.bytes)
+
+}
+
+object ExecuteContractFunctionTransaction {
+
+  val MaxDescriptionSize = 140
+  val maxDescriptionStringSize: Int = base58Length(MaxDescriptionSize)
+
+  def parseTail(bytes: Array[Byte]): Try[ExecuteContractFunctionTransaction] = Try {
+    (for {
+      contractId <- ContractAccount.fromBytes(bytes.slice(0, ContractAccount.AddressLength))
+      funcIdx = Shorts.fromByteArray(bytes.slice(ContractAccount.AddressLength, ContractAccount.AddressLength + 2))
+      (dataBytes, dataEnd) = Deser.parseArraySize(bytes, ContractAccount.AddressLength + 2)
+      data <- DataEntry.parseArrays(dataBytes)
+      (description, descriptionEnd) = Deser.parseArraySize(bytes, dataEnd)
+      fee = Longs.fromByteArray(bytes.slice(descriptionEnd, descriptionEnd + 8))
+      feeScale = Shorts.fromByteArray(bytes.slice(descriptionEnd + 8, descriptionEnd + 10))
+      timestamp = Longs.fromByteArray(bytes.slice(descriptionEnd + 10, descriptionEnd + 18))
+      proofs <- Proofs.fromBytes(bytes.slice(descriptionEnd + 18, bytes.length))
+      tx <- ExecuteContractFunctionTransaction.createWithProof(contractId, funcIdx,
+        data, description, fee, feeScale, timestamp, proofs)
+    } yield tx).fold(left => Failure(new Exception(left.toString)), right => Success(right))
+  }.flatten
+
+  def createWithProof(contractId: ContractAccount,
+                      funcIdx: Short,
+                      data: Seq[DataEntry],
+                      attachment: Array[Byte],
+                      fee: Long,
+                      feeScale: Short,
+                      timestamp: Long,
+                      proofs: Proofs): Either[ValidationError, ExecuteContractFunctionTransaction] =
+    if (attachment.length > MaxDescriptionSize) {
+      Left(ValidationError.TooBigArray)
+    } else if (fee <= 0) {
+      Left(ValidationError.InsufficientFee)
+    } else if (feeScale != 100) {
```

Comment:
is 100 configured somewhere?

Reply:
this is related with fee calculation system. in current version we only accept 100. so it was checked in every transaction

Reply:
should we define a constant for this number?